src/master/develop/reading.md

```diff
@@ -367,14 +382,95 @@ const IMMUDB_PWD = 'immudb'
 const cl = new ImmudbClient({ host: IMMUDB_HOST, port: IMMUDB_PORT });
 
 (async () => {
-	await cl.login({ user: IMMUDB_USER, password: IMMUDB_PWD })
-	const { id } = await cl.set({ key: 'key', value: 'value' })
-	
-	const verifiedTxByIdReq: Parameters.VerifiedTxById = { tx: id }
-	const verifiedTxByIdRes = await cl.verifiedTxByID(verifiedTxByIdReq)
-	console.log('success: verifiedTxByID', verifiedTxByIdRes)
+    await cl.login({ user: IMMUDB_USER, password: IMMUDB_PWD })
+    const { id } = await cl.set({ key: 'key', value: 'value' })
+    
+    const verifiedTxByIdReq: Parameters.VerifiedTxById = { tx: id }
+    const verifiedTxByIdRes = await cl.verifiedTxByID(verifiedTxByIdReq)
+    console.log('success: verifiedTxByID', verifiedTxByIdRes)
 })()
 ```
+
+:::
+
+::: tab .Net
+This feature is not yet supported or not documented.
+Do you want to make a feature request or help out? Open an issue on [.Net sdk github project](https://github.com/codenotary/immudb4dotnet/issues/new)
+:::
+
+::: tab Others
+If you're using another development language, please refer to the [immugw](/master/immugw/) option.
+:::
+
+::::
+
+### Conditional writes
+
+Immudb can check additional preconditions before the write operation is made.
+Precondition is checked atomically with the write operation.
+It can be then used to ensure consistent state of data inside the database.
+
+Following preconditions are supported:
+
+* MustExist - precondition checks if given key exists in the database
+* MustNotExist - precondition checks if given key does not exist in the database
+* NotModifiedAfterTX - precondition checks if given key was not modified after given transaction id
+
+*Note: Checks do not consider logical deletion of entries.
+E.g. if an entry was logically deleted, it will still pass the `MustExist` precondition check.*
+
+Entries used in precondition do not have to overlap with keys to be written.
```

Comment:
what about the case of ensuring it's the first write to a key e.g. mustNotExist key1, set key1, value1?